pkg/apis/events/v1alpha1/cloudpubsubsource_validation_test.go

```diff
@@ -180,7 +180,16 @@ func TestCloudPubSubSourceCheckValidationFields(t *testing.T) {
 		"invalid GCP service account": {
 			spec: func() CloudPubSubSourceSpec {
 				obj := pubSubSourceSpec.DeepCopy()
-				obj.ServiceAccount = &invalidServiceAccountName
+				obj.ServiceAccount = invalidServiceAccountName
+				return *obj
+			}(),
+			error: true,
+		},
+		"have GCP service account and secret in the same time": {
```

Comment:
and it would be great if you can add a UT for the empty secret as well, so that we do not regress